core/shared/src/main/scala-3/monocle/internal/focus/features/selectfield/SelectFieldGenerator.scala

```diff
@@ -18,13 +18,29 @@ private[focus] trait SelectFieldGenerator {
     def generateSetter(from: Term, to: Term): Term =
       Select.overloaded(from, "copy", fromTypeArgs, NamedArg(fieldName, to) :: Nil) // o.copy(field = value)
 
-    (fromType.asType, toType.asType) match {
-      case ('[f], '[t]) =>
-        '{
-          Lens.apply[f, t]((from: f) => ${ generateGetter('{ from }.asTerm).asExprOf[t] })((to: t) =>
-            (from: f) => ${ generateSetter('{ from }.asTerm, '{ to }.asTerm).asExprOf[f] }
-          )
-        }.asTerm
+    action.selectType match {
+      case SelectType.CaseClassField =>
+        (fromType.asType, toType.asType) match {
+          case ('[f], '[t]) =>
+            '{
+              Lens.apply[f, t]((from: f) => ${ generateGetter('{ from }.asTerm).asExprOf[t] })((to: t) =>
+                (from: f) => ${ generateSetter('{ from }.asTerm, '{ to }.asTerm).asExprOf[f] }
+              )
+            }.asTerm
+        }
+      case SelectType.PublicField =>
+        (fromType.asType, toType.asType) match {
+          case ('[f], '[t]) =>
+            '{ Getter.apply[f, t]((from: f) => ${ generateGetter('{ from }.asTerm).asExprOf[t] }) }.asTerm
+        }
+      case SelectType.VirtualField =>
+        (fromType.asType, toType.asType) match {
+          case ('[f], '[t]) =>
+            '{
+              Getter.apply[f, t]((from: f) => ${ generateGetter('{ from }.asTerm).appliedToArgs(List()).asExprOf[t] })
```

Comment:
To be honest, we should stamp the symbol in the parser so we don't do the resolution a second time - this currently doesn't work when we have clashing names like most Java classes.